src/core/qgscoordinateutils.cpp

```diff
@@ -142,6 +142,40 @@ QString QgsCoordinateUtils::formatExtentForProject( QgsProject *project, const Q
                                           QgsCoordinateUtils::formatCoordinateForProject( project, p2, destCrs, precision ) );
 }
 
+double QgsCoordinateUtils::degreeToDecimal( const QString &string, bool *ok, bool *isEasting )
+{
+  const QString negative( QStringLiteral( "swSW" ) );
+  const QString easting( QStringLiteral( "eEwW" ) );
+  double value = 0.0;
+  bool okValue = false;
+
+  if ( ok )
+  {
+    *ok = false;
+  }
+  else
+  {
+    ok = &okValue;
+  }
+
+  QRegularExpression degreeWithSuffix( QStringLiteral( "^([0-9\\-\\.]*)\\s*([NSEW])$" ) );
+  QRegularExpressionMatch match = degreeWithSuffix.match( string.trimmed() );
```

Comment:
```suggestion
  QRegularExpression degreeWithSuffix( QStringLiteral( "^\s*([0-9\\-\\.]*)\\s*([NSEW])\s*$" ) );
  QRegularExpressionMatch match = degreeWithSuffix.match( string );
```

saves an extra string allocation